awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -556,6 +563,24 @@ def create_if_needed(self, existing_item, new_item, endpoint, on_create=None, it
         else:
             self.exit_json(**self.json_output)
 
+    # We need to be able to recursevly step through fields in the case of inputs for credentials.
+    # They are dicts and we can't just compare them at the top level because the dict returned from the tower api will have fields like $encrypted$
```

Comment:
Again, this doesn't address the case of $encrypted$, so lose that, but the rest of the comment is valuable.

Reply:
Comment updated.